agent/provider/src/hardware.rs

```diff
@@ -499,6 +499,7 @@ fn partition_space<P: AsRef<Path>>(path: P) -> Result<u64, Error> {
         use nix::sys::statvfs::statvfs;
         let stat =
             statvfs(path.as_os_str()).map_err(|e| sys_info::Error::General(e.to_string()))?;
+        #[allow(clippy::unnecessary_cast)]
```

Comment:
why allow instead of removing the cast?

Reply:
This is because on mac & linux these values differ :<

Reply:
cool